tests/main/snap-userd-socket-activation/task.yaml

```
@@ -0,0 +1,47 @@
+summary: Check that root can activate userd via socket activation
+
+systems:
+    - -ubuntu-core-*
+    - -ubuntu-14.04-*
+    - -amazon-linux-2-*
+    - -centos-7-*
+
+environment:
+    TEST_UID: $(id -u test)
+    USER_RUNTIME_DIR: /run/user/${TEST_UID}
+
+prepare: |
+    mkdir -p "$USER_RUNTIME_DIR"
+    chmod u=rwX,go= "$USER_RUNTIME_DIR"
+    chown test:test "$USER_RUNTIME_DIR"
+    systemctl start "user@${TEST_UID}.service"
+
+restore: |
+    systemctl stop "user@${TEST_UID}.service"
+    rm -rf "${USER_RUNTIME_DIR:?}"/* "${USER_RUNTIME_DIR:?}"/.[!.]*
+
+execute: |
+    systemctl_user() {
+        su -l -c "XDG_RUNTIME_DIR=\"${USER_RUNTIME_DIR}\" systemctl --user $*" test
+    }
+    echo "Initially snap userd is not running"
+    ! systemctl_user is-active snap-userd.service
+
+    echo "However its REST API socket exists"
+    [ -S "${USER_RUNTIME_DIR}/snap-session.socket" ]
```

Nitpick: this would read nicer as `test`:
```suggestion
    test -S "${USER_RUNTIME_DIR}/snap-session.socket"
```